crates/vhost-device-gpio/src/backend.rs

```diff
@@ -151,9 +181,17 @@ fn start_backend<D: 'static + GpioDevice + Send + Sync>(args: GpioArgs) -> Resul
             // threads, and so the code uses unwrap() instead. The panic on a thread won't cause
             // trouble to other threads/guests or the main() function and should be safe for the
             // daemon.
-            let device = D::open(device_num).map_err(Error::CouldNotOpenDevice)?;
-            let controller =
-                GpioController::<D>::new(device).map_err(Error::CouldNotCreateGpioController)?;
+            let controller = match cfg {
+                GpioDeviceType::PhysicalDevice(device_num) => match D::open(device_num) {
+                    Ok(device) => match GpioController::<D>::new(device) {
+                        Ok(controller) => controller,
+                        Err(error) => return Err(Error::CouldNotCreateGpioController(error)),
+                    },
+                    Err(error) => return Err(Error::CouldNotOpenDevice(error)),
+                },
+                _ => return Err(Error::DeviceUnimplemented),
+            };
```

Comment:
After discussing it on IRC. I take this back. This just does not work since it would require dynamic dispatch. And even if we would add the boilerplate code to implement GpioDevice for Box<GpioDevice>,  the `open() -> T` contract of GpioDevice prevents dropping the Sized qualifier. So it just does not work without doing changes out of scope of this PR.